### ISSUE TITLE
Crashed while deleting a bunch of resources

### ISSUE THREAD

**Issue by serbrech** [lawrencegripper/azbrowse#225]
Delete followed by CTRL+Y on a bunch of RG in a row and this happened
Pasting this quickly here to give you data before I have a change to look further.

```log                                                                                                                                               ➜  release git:(master) azbrowse 
2020/03/03 13:39:36 Current version is the latest
Loading db ...
Loading db complete

	/usr/local/go/src/internal/poll/fd_unix.go:169 +0x22b
net.(*netFD).Read(0xc000c30600, 0xc00133c000, 0x40e8, 0x40e8, 0x203000, 0x2d2b688, 0x5)
	/usr/local/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc0000a86b8, 0xc00133c000, 0x40e8, 0x40e8, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:184 +0x68
crypto/tls.(*atLeastReader).Read(0xc0003220a0, 0xc00133c000, 0x40e8, 0x40e8, 0xc000d799b8, 0x100de3e, 0xc000d79970)
	/usr/local/go/src/crypto/tls/conn.go:780 +0x60
bytes.(*Buffer).ReadFrom(0xc000109ad8, 0x211ec20, 0xc0003220a0, 0x1009a85, 0x19df4a0, 0x1aa7000)
Delete followed by CTRL + Y a dozen time in a row and this occured.

/usr/local/go/src/bytes/buffer.go:204 +0xb4
crypto/tls.(*Conn).readFromUntil(0xc000109880, 0x2f61058, 0xc0000a86b8, 0x5, 0xc0000a86b8, 0x203000)
	/usr/local/go/src/crypto/tls/conn.go:802 +0xec
crypto/tls.(*Conn).readRecordOrCCS(0xc000109880, 0x0, 0x0, 0xc0004338c0)
	/usr/local/go/src/crypto/tls/conn.go:609 +0x124
crypto/tls.(*Conn).readRecord(...)
	/usr/local/go/src/crypto/tls/conn.go:577
crypto/tls.(*Conn).Read(0xc000109880, 0xc000d9a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:1255 +0x161
bufio.(*Reader).Read(0xc0005d37a0, 0xc000649458, 0x9, 0x9, 0xc000d79d10, 0x0, 0x12ed642)
	/usr/local/go/src/bufio/bufio.go:226 +0x26a
io.ReadAtLeast(0x211ea40, 0xc0005d37a0, 0xc000649458, 0x9, 0x9, 0x9, 0xc00008a050, 0x0, 0x211ee40)
	/usr/local/go/src/io/io.go:310 +0x87
io.ReadFull(...)
	/usr/local/go/src/io/io.go:329
net/http.http2readFrameHeader(0xc000649458, 0x9, 0x9, 0x211ea40, 0xc0005d37a0, 0x0, 0x0, 0xc0004361b0, 0x0)
	/usr/local/go/src/net/http/h2_bundle.go:1477 +0x87
net/http.(*http2Framer).ReadFrame(0xc000649420, 0xc0004361b0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/h2_bundle.go:1735 +0xa1
net/http.(*http2clientConnReadLoop).run(0xc000d79fb8, 0x12c758d, 0xc000074a20)
	/usr/local/go/src/net/http/h2_bundle.go:8175 +0x8e
net/http.(*http2ClientConn).readLoop(0xc000467980)
	/usr/local/go/src/net/http/h2_bundle.go:8103 +0xa3
created by net/http.(*http2Transport).newClientConn
	/usr/local/go/src/net/http/h2_bundle.go:7162 +0x62f
```

**Comment by lawrencegripper** [comment]
Thanks, this looks like an interesting one as it's not been caught by the error recovery stuff. 

May be related to #220 and #221 will take a look.

**Comment by lawrencegripper** [comment]
Closing this out now as should have been fixed up by #221. If it reoccurs let me know and we can reopen and take a look at getting a test to repro the issue.
